alacritty_terminal/src/term/mod.rs

```diff
@@ -830,6 +836,29 @@ impl<T> Term<T> {
         } else {
             self.grid.update_history(config.scrolling.history() as usize);
         }
+
+        self.update_blinking_cursor_context();
+    }
+
+    /// Depending on current mode and cursor configuration, decide whether to start or stop
+    /// blinking the cursor.
+    ///
+    /// Typically used after a configuration update or vi_mode toggle.
+    pub fn update_blinking_cursor_context(&self)
```

Comment:
I'm questioning the API ergonomics of how this is supposed to work. The terminal instructs the UI when blinking should start or end, then the UI tells the terminal to tell the UI to update it about how things should work? That seems insanely convoluted.

Why can't the UI just figure this out itself?

Reply:
I didn't find a better way to do that, but that's my first contribution so if you think that's convoluted there must be an other way. I'll look into it and ask question on IRC if needed.